### DISCUSSION TITLE
Donating a Maven OpenTelemetry extension to the OpenTelemetry project?

### DISCUSSION THREAD

**Discussion by cyrille-leclerc** [open-telemetry/community#832] [Community Discussions]
Dear community,

We would like to donate to the OpenTelemetry project a [Maven OpenTelemetry extension](https://github.com/elastic/opentelemetry-maven-extension) that we have created to observe the execution of Maven builds.

This Maven instrumentation integrates with other OpenTelemetry capable CI/CD tools such as Jenkins pipelines ([Jenkins OpenTelemetry plugin](https://github.com/jenkinsci/opentelemetry-plugin)), Ansible deployments ([Ansible OpenTelemetry plugin](https://github.com/ansible-collections/community.general/pull/3091)) or the [otel-cli](https://github.com/equinix-labs/otel-cli).
 
Here is an example of distributed trace providing visibility of a Jenkins / Maven CI pipeline.

![](https://raw.githubusercontent.com/cyrille-leclerc/opentelemetry-maven-extension/main/docs/images/jenkins-maven-execution-trace-jaeger.png)

It's very straightforward to test the extension on any existing Maven project as we have documented [here](https://github.com/elastic/opentelemetry-maven-extension#getting-started).

We have very positive feedback from CI/CD practitioners when we show the capabilities and the extension is already used by Elastic on its production CI platform in conjunction with the Jenkins OpenTelemetry plugin.

We would like to donate the Maven OpenTelemetry extension to the OpenTelemetry project / CNCF foundation because we feel that it would be the best place to develop this integration and because we see the opportunity to create an entire ecosystem of  OpenTelemetry capable CI/CD tools.

Note that we preferred to build this integration as a Maven extension rather than as an auto instrumentation that would be part of the [OpenTelemetry Java agent](https://github.com/open-telemetry/opentelemetry-java-instrumentation) because the Maven community is used to define its builds adding plugins and extension to build definitions (pom.xml) rather than wiring a Java auto-instrumentation agent when launching the "mvn" command.

**Would the OpenTelemetry project be interested in hosting such [Maven OpenTelemetry integration](https://github.com/elastic/opentelemetry-maven-extension)?**

**Comment by trask** [comment]
hi @cyrille-leclerc! this looks like a great contribution for https://github.com/open-telemetry/opentelemetry-java-contrib. I don't think this would be required to go through the [formal donation process](https://github.com/open-telemetry/community/blob/main/CONTRIBUTING.md#donations) if you want to jump straight to opening an issue/discussion over there?

**Reply by kenfinnigan** [reply]
I've had a similar idea for a while and hadn't had the time to implement it so I'm all for it being donated!

**Comment by cyrille-leclerc** [comment]
Thanks @trask  and @kenfinnigan . I have created:
- https://github.com/open-telemetry/opentelemetry-java-contrib/issues/78
